pkg/collect/host_run.go

```diff
@@ -31,20 +37,67 @@ func (c *CollectHostRun) IsExcluded() (bool, error) {
 }
 
 func (c *CollectHostRun) Collect(progressChan chan<- interface{}) (map[string][]byte, error) {
+	var (
+		cmdOutputTempDir         string
+		cmdInputTempDir          string
+		bundleOutputRelativePath string
+	)
+
 	runHostCollector := c.hostCollector
+	collectorName := runHostCollector.CollectorName
+	if collectorName == "" {
+		collectorName = "run-host"
+	}
 
 	cmd := exec.Command(runHostCollector.Command, runHostCollector.Args...)
 
-	var stdout, stderr bytes.Buffer
-	cmd.Stdout = &stdout
-	cmd.Stderr = &stderr
-
-	runInfo := HostRunInfo{
+	runInfo := &HostRunInfo{
 		Command:  cmd.String(),
 		ExitCode: "0",
 	}
 
-	err := cmd.Run()
+	err := c.processEnvVars(cmd)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse env variable")
+	}
+
+	// if we choose to save result for the command run
+	if runHostCollector.OutputDir != "" {
+		cmdOutputTempDir, err = os.MkdirTemp("", runHostCollector.OutputDir)
+		defer os.RemoveAll(cmdOutputTempDir)
+		if err != nil {
+			return nil, errors.New(fmt.Sprintf("failed to created temp dir for: %s", runHostCollector.OutputDir))
+		}
+		cmd.Env = append(cmd.Env,
+			fmt.Sprintf("TS_WORKSPACE_DIR=%s", cmdOutputTempDir),
+		)
+	}
+
+	if runHostCollector.Input != nil {
+		cmdInputTempDir, err = os.MkdirTemp("", "input")
+		defer os.RemoveAll(cmdInputTempDir)
+		if err != nil {
+			return nil, errors.New("failed to created temp dir for host run input")
+		}
+		for inFilename := range runHostCollector.Input {
+			if strings.Contains(inFilename, "/") {
```

Comment:
It seems like you do not do anything with the input files besides checking that they have no `/`

Reply:
correct. i used to also collect the input file into the bundle but talking with @mhrabovcin we think that the input files can be sensitive to anyone that has access to the bundle while the bundle creator has the input anyways.

Reply:
Isn't the idea to store them in `TS_INPUT_DIR` then delete them once done? If I understood correctly what you want to avoid is having them end up in the support bundle, correct?

Reply:
introduced a commit that is not needed -- reverted that commit now.